rewrite-core/src/main/kotlin/com/netflix/rewrite/parse/OracleJdkParserVisitor.kt

```diff
@@ -1143,8 +1143,10 @@ class OracleJdkParserVisitor(val path: Path, val source: String): TreePathScanne
                                     )
                                 }
 
+                        val symType = sym.type as com.sun.tools.javac.code.Type.ClassType
                         Type.Class.build(sym.className(), fields, type(type.supertype_field, stack.plus(sym)).asClass(),
-                                type.typarams_field?.map { tParam -> type(tParam, stack.plus(sym), shallow = true) }?.filterNotNull() ?: emptyList())
+                                type.typarams_field?.mapNotNull { tParam -> type(tParam, stack.plus(sym), shallow = true) } ?: emptyList(),
+                                symType.interfaces_field?.mapNotNull { i -> type(i, stack.plus(sym), shallow = true) } ?: emptyList())
```

Comment:
I'm not entirely sure why the `type.interfaces_field` yields `null`, but the nested `type.tsym.type.interfaces_field` has the information we want.

Also, it would be nice to plumb through `all_interfaces_field`, but for some reason in my test case this field was always `null` (even the nested version).

Reply:
I know the feeling. The Sun AST has some really dark corners :)